### ISSUE TITLE
Doc: Docs are a bit misleading

### ISSUE THREAD

**Issue by vfonic** [rubocop-hq/rubocop#6878]
First of all, I love rubocop! 👍 ❤️ 
Thank you!

## tl;dr

The bugs:

[1] rubocop CLI accepts several arguments, but those don't map to the "equally" named configuration parameters in YAML file
Solution: Ensure CLI arguments _intuitively_ map to `.rubocop.yml` configuration options

Example:

```sh
$ rubocop --rails --safe-auto-correct --display-cop-names
```

Should map to something like:

```yaml
AllCops:
  DisplayCopNames: true
  Rails: true
  SafeAutoCorrect: true
```

[2] Some parts of YAML configuration report a warning when there's a non-recognized configuration option provided, some don't. For example:

```yaml
AllCops:
  Rails: true
```

...reports a warning, while this doesn't:

```yaml
AllCops:
  AutoCorrect:
    Whatevah: true
```

Solution: Ensure all non-existing config options are reported as warnings/errors.

Example:

```yaml
AllCops:
  AutoCorrect:
    Whatevah: true
```

Reports an error/warning.

## Is your feature request related to a problem?

I'm trying to setup rubocop for a new project. I want to use the latest released version of rubocop. It's a rails project and I also use rspec.

I don't know...where do I even start?

Maybe it's the best to write my "user scenario", exactly (more-less) how it happened:

1. I went to: https://github.com/rubocop-hq/rubocop
There's installation section: https://github.com/rubocop-hq/rubocop#installation
So I added rubocop to my `Gemfile` like it's explained there: `gem 'rubocop', require: false`
I was wondering why the gem is not added to `:development' group, but couldn't find any explanation. Maybe there's rubocop-rails gem I can use to figure that out.
2. I went to search for `rubocop-rails`. This yielded three relevant results: link to this repo, link to https://github.com/toshimaru/rubocop-rails, which made me spend some time and finally give up again and rubocop-rails repo that looked like the thing I needed: https://github.com/rubocop-hq/rubocop-rails. However, there are no instructions on how to install or use it.
3. I continued reading documentation: https://docs.rubocop.org/en/latest/basic_usage/
I realized I could enforce command line options through `.rubocop.yml`. So I decided I'll use

```
AllCops:
  DisplayCopNames: true
  DisplayStyleGuide: true
  ExtraDetails: true
  Rails: true
  SafeAutoCorrect: true
```

These are all taken from:
https://docs.rubocop.org/en/latest/basic_usage/
Command-line flags

That gives me the following warnings:

> Warning: AllCops does not support Rails parameter.
> Warning: AllCops does not support SafeAutoCorrect parameter.

So CLI arguments don't really map exactly to YAML arguments, but there's no explanation on how to enable Rails cop(s).[1]

4. Check https://docs.rubocop.org/en/latest/cops_rails/, nothing
4.b) Check https://docs.rubocop.org/en/latest/configuration/, nothing
4.c) Check https://docs.rubocop.org/en/latest/cops/#rails, BINGO!

5. Configure rubocop-rspec. That was a breeze (how to install, how to configure/setup, how to run): https://github.com/rubocop-hq/rubocop-rspec#installation
Maybe something similar could be added to rubocop Readme or docs? After all rails is so widely used that rubocop even has a "native" support for its cops. 

6. I still have SafeAutoCorrect which doesn't work. So I decide I'll use my "webpack jedi skills" (read: I'll shuffle things around in config until it magically hopefully works)
So I convert:

```
AllCops:
  SafeAutoCorrect: true
```

to:

```
AllCops:
  AutoCorrect:
    Safe: true
    Mama: true
```

`Mama` is there to ensure I get an error if I type in a config option that doesn't exist.
There's no error. I'm confused, I'm sure I got the error (warning) for `AllCops:Rails:true`. I just expected that, if I get error once, I'll get it for nested configs as well. So `AllCops:AutoCorrect:Safe` _maybe_ works... Running `rubocop --safe-auto-correct` proves that it doesn't. 🤷‍♂️ [2]

After continuing my search, the only thing that resembles what I need is adding `Safe: true` and `SafeAutoCorrect` to all cops, one by one. Or passing it to CLI every time. Is there a better way?

Thank you!

**Comment by vfonic** [comment]
Bump! Any thoughts? Am I overthinking this?

**Comment by bbatsov** [comment]
You've raised some very good points, but unfortunately I'm crazy busy these days and I don't have time to properly respond. Long-story short - there's a lot of inconsistencies all over the project and I'd definitely would like to reduce those before 1.0. We should certainly improve the configuration validation, no argument from me. 

As for `rubocop-rails` - it's a WIP project right now, that's scheduled to be completed before 1.0 as well. (but I guess you figured this out by now) I'll try to find some time to revisit your questions in more details during the next week.

**Comment by vfonic** [comment]
@bbatsov Thank you very much for your time and work! ❤️ 

I just wanted to raise some points about some difficulties when using the library. I see that you are aware of these issues, so don't spend too much time answering to me. 👍 
I think the best is to close this issue for now. I'm looking forward to v1.0!
